pkg/secret/secrets.go

```diff
@@ -23,3 +25,24 @@ func LoadSecretData(apiReader client.Reader, secretName, namespace, dataKey stri
 	}
 	return string(retStr), nil
 }
+
+func ReadBasicAuthSecret(apiReader clientv1.SecretInterface, secretName string) (string, string, error) {
+	secret, err := apiReader.Get(context.TODO(), secretName, metav1.GetOptions{})
+	var username, password string
+	if err != nil {
+		return "", "", err
+	}
+
+	for key, value := range secret.Data {
```

Comment:
This switch will not return an error in case either "username" or "password" is missing, we should catch that case and return an actionable error.

Reply:
That is a good point!